### ISSUE TITLE
Unit-Test example does not work with jasmine 2.5.1 only 2.4.1....

### ISSUE THREAD

**Issue by Nefarion** [angular/angular#11658]
**I'm submitting a ...**  (check one with "x")

```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior** 
The code example from the "Advanced Documentation -> Testing" section is written with jasmine 2.4.1, if you install jasmine 2.5.0 or 2.5.1 then it breaks.

[link to example](https://angular.io/resources/live-examples/testing/ts/1st-specs.plnkr.html)
**Expected behavior**
Should work with jasmine 2.5.0 and 2.5.1

**Reproduction of the problem**
Just change the line 15, 16 and 17 from ...../2.4.1/... to .../2.5.1/... and it breaks, complaining about a property missing on an object.

**What is the motivation / use case for changing the behavior?**
2.5.1 fixed some errors in jasmine when using time features.
- **Angular version:** 2.0.0
- **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ] 
  Chrome 53.0.2785.116 m (64-bit)

Greetings!

**Comment by pkozlowski-opensource** [comment]
> The code example from the "Advanced Documentation -> Testing" section is written with jasmine 2.4.1, if you install jasmine 2.5.0 or 2.5.1 then it breaks.

Could you please elaborate on "it breaks"? What are the errors? Did you investigate the root cause?

**Comment by Nefarion** [comment]
The specific error is:
```zone.js@0.6.23?main=browser:346 TypeError: Cannot read property 'spies' of undefined```

I cannot find why, but i think it might be related to 2.5.0 "fixes" to spies and their cleanup, and how ```zone.js/jasmine-patch.js``` is patching the onComplete call

**Comment by ericmartinezr** [comment]
https://github.com/angular/angular/issues/11323

**Comment by Nefarion** [comment]
@ericmartinezr Thanks, i was looking for an issue but i could not find it.

**Comment by vytautas-pranskunas-** [comment]
Any progress on this?

**Comment by DzmitryShylovich** [comment]
I think it should be closed and re-opened in https://github.com/angular/angular.io as it's documentation issue.

**Comment by SjoerdL** [comment]
I also found this issue when trying to work with Jasmine 2.5.2. I also noticed that an issue and work-around were posted on the Jasmine page, which might be useful for other people looking into this: https://github.com/jasmine/jasmine/issues/1220

**Comment by mLaird** [comment]
I just started following the Testing section in the docs using Jasmine 2.5.2 and am using 4.0.0-beta.0. When I do npm test, I get hundreds of lines of errors in @angular, typings, @types, and a few others - followed by "Test failed". My project structure is the same as the Getting Started app, but I have built my own app.  The test that fails is the 1st Karma test of true=true
    
    expect(true).toBe(true)

I have tried excluding in karma.config with

    exclude: [ 'node_modules/**/*spec.js', 'typings/**/index.d.ts' ],

but I still get errors in files starting with those folders.

I suspect documentation shortcomings.  If I could see karma.conf.js, karma-test-shim.js systemjs.config.js and systemjs.config.extras.js  that are used in the Test section docs, I might be able to figure out changes to my files to try and solve my problem.  As it is, I am looking at large files and huge error reports and don't know where to start, or even what I should Google to learn more.
